libs/features/mas/web-components/src/utils.js

```diff
@@ -13,141 +11,20 @@ export function debounce(func, delay) {
-    if (!first) return null;
-    let ancestor = first.parentElement;
-    if (!rest.length) return ancestor;
-    while (ancestor) {
-        if (rest.every((element) => ancestor.contains(element))) {
-            return ancestor;
-        }
-        ancestor = ancestor.parentElement;
-    }
-    return null;
-}
-
-/**
- * @param {Commerce.Placeholder} placeholder
- * @returns {boolean}
- */
-export function isRegularPrice(placeholder) {
-    if (placeholder.isInlinePrice) {
-        const { template } = placeholder.dataset;
-        if (template === 'price' || !template) {
-            return true;
-        }
-    }
-    return false;
-}
-
-/**
- * Joins an array of string `tokens`,
- * filtering out `falsy` values and replacing non-word characters in each token.
- * @param { string[] } tokens
- * Array of strings to be joined.
- * @param { string } delimiter
- * Delimiter used to join the tokens and replace non-word characters.
- */
-const joinTokens = (tokens, delimiter) =>
-    tokens
-        .flatMap((token) => token?.split?.(/\W+/g))
-        .filter((token, index) => token && (index > 0 || token !== NAMESPACE))
-        .join(delimiter);
-
-/**
- * Creates a namespaced CSS class name.
- * Replaces any sequence of non-word characters with a single hyphen.
- * @param {...string} args - Components of the class name.
- * @returns {string} - The namespaced and cleaned CSS class name.
- */
-export const makeCssClassName = (...args) =>
-    `${NAMESPACE}-${joinTokens(args, '-')}`;
-
-/**
- * Creates a namespaced event type.
- * Replaces any sequence of non-word characters with a single colon.
- * @param {...string} args - Components of the event type.
- * @returns {string} The namespaced and cleaned event type.
- */
-export const makeEventTypeName = (...args) =>
-    `${NAMESPACE}:${joinTokens(args, ':')}`;
-
-/**
- * @param {Commerce.Offers} offers
- * @param {Commerce.filterOffer} filter
- */
-export function showOffers(offers, filter = () => true) {
-    let i = 0;
-    for (const offer of offers.values()) {
-        offer.container?.classList.toggle(
-            CLASS_NAME_HIDDEN,
-            !filter(offer, i++),
-        );
-    }
-}
-
-/**
- * Returns the text nodes of the given element.
- * @param {HTMLElement} element
- * @returns array of text nodes
- */
-export function getTextNodes(element) {
-    const textNodes = [];
-    function findTextNodes(node) {
-        if (node.nodeType === Node.TEXT_NODE) {
-            textNodes.push(node);
-        } else {
-            node.childNodes.forEach(findTextNodes);
-        }
-    }
-
-    findTextNodes(element);
-    return textNodes;
-}
-
 /**
  * Helper function to create an element with attributes
  * @param {string} tag
  * @param {Object} attributes
- * @param {*} innerHTML
+ * @param {*} content
  * @returns {HTMLElement}
  */
-export function createTag(tag, attributes = {}, innerHTML) {
+export function createTag(tag, attributes = {}, content) {
```

Comment:
This function looks similar to Milo's. Are we recreating it to decouple the 2 codebases?

Reply:
I don't want to pull Milo's utils.js in mas.js that we ship to non Milo consumers as well.